addons/mrp/report/mrp_report_mo_overview.py

```diff
@@ -751,23 +759,38 @@ def _add_origins_to_forecast(self, forecast_lines):
                     # Remove 'in_transit' for MTO replenishments
                     line['in_transit'] = False
                     move_origin_qty = move_origin.product_uom._compute_quantity(move_origin.product_uom_qty, line['uom_id'])
+                    if not origin_qty.get(doc_origin):
+                        origin_qty[doc_origin] = move_origin_qty
                     # Move quantity matches forecast, can add origin to the line
-                    if float_compare(line['quantity'], move_origin_qty, precision_rounding=line['uom_id'].rounding) == 0:
+                    if float_compare(line['quantity'], origin_qty[doc_origin], precision_rounding=line['uom_id'].rounding) == 0:
```

Comment:
I'm not aware of the case where this could happen, but originally we could guarantee that `move_origin_qty` was in the same `uom` as `line['quantity']` due to the conversion above.

This is technically no longer the case.